sale_coupon_portal/controllers/portal.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, http
+from odoo.http import request
+from odoo.osv import expression
+
+from odoo.addons.portal.controllers.portal import CustomerPortal, pager as portal_pager
+
+
+class PortalCoupon(CustomerPortal):
+    def _prepare_home_portal_values(self):
+        values = super()._prepare_home_portal_values()
+        coupon_count = (
+            request.env["sale.coupon"].sudo().search_count(self._get_coupons_domain())
+        )
+        values["coupon_count"] = coupon_count
+        return values
+
+    def _get_coupons_domain(self):
+        return [("partner_id", "child_of", request.env.user.commercial_partner_id.id)]
```

Comment:
To be able to work like this, you'd need to depend on [sale_coupon_apply_commercial_partner](https://github.com/OCA/sale-promotion/pull/105) otherwise you wouldn't be able to validate the coupons that don't belong to the order's partner.

The options are:

- A bridge module
- This depending on that
- That depending on this

What do you think @sergio-teruel ?